### PR TITLE
docs(config): rename optimization.noEmitOnErrors to optimization.emitOnErrors

### LINKED ISSUE #3862: Document webpack change: deprecate optimization.noEmitOnErrors in favor of optimization.emitOnErrors
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/11193

---

**What kind of change does this PR introduce?**
deprecation

**Did you add tests for your changes?**
no

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
`optimization.noEmitOnErrors` -> `optimization.emitOnErrors`

### PR DESCRIPTION
**Pull request by snitin315**
Resolve #3862 

[1]: https://cla.js.foundation/webpack/webpack.js.org
[2]: https://webpack.js.org/contribute/writers-guide/
[3]: https://webpack.js.org/contribute/#pull-requests

### CODE REVIEW

**Comment by snitin315** [comment]
I will fix the CI.

File: src/content/configuration/optimization.md [hunk 1]
```diff
@@ -145,19 +145,19 @@ module.exports = {
 };
 ```
 
-## `optimization.noEmitOnErrors`
+## `optimization.emitOnErrors`
 
 `boolean`
```
Comment (EugeneHlushko) [inline src/content/configuration/optimization.md:150]:
```suggestion
`boolean = false`
```

**Review by EugeneHlushko — DISMISSED** [review]
You had renamed the heading therefore the anchor changed, now pages linking to deleted/changed anchor need fixing... this info is available on travis and could be triggered manually after running a build

```
✖ FAIL fragment-check dist/plugins/index.html --> /configuration/optimization/#optimizationnoemitonerrors

  | operator: fragment-check

  | expected: id="optimizationnoemitonerrors"

  |       at: dist/plugins/index.html:104:34 <a href="/configuration/optimization/#optimizationnoemitonerrors">...</a>

  

  ✖ FAIL fragment-check dist/plugins/printable/index.html --> /configuration/optimization/#optimizationnoemitonerrors

  | operator: fragment-check

  | expected: id="optimizationnoemitonerrors"

  |       at: dist/plugins/printable/index.html:3208:188 <a href="/configuration/optimization/#optimizationnoemitonerrors">...</a>
```

File: src/content/configuration/optimization.md [hunk 2]
```diff
@@ -155,19 +155,19 @@ module.exports = {
...
```
Comment (snitin315) [inline src/content/configuration/optimization.md:160]:
```suggestion
`boolean = false`
```

**Comment by EugeneHlushko** [comment]
@snitin315 still failing

**Comment by snitin315** [comment]
Looking into it.

**Comment by snitin315** [comment]
I don't know how to fix this 😞 , can someone help?

File: src/content/configuration/optimization.md [hunk 3]
```diff
@@ -155,19 +155,19 @@ module.exports = {
 };
 ```
 
-## `optimization.noEmitOnErrors`
+## `optimization.emitOnErrors`
```
Comment (evilebottnawi) [inline src/content/configuration/optimization.md:158]: Our docs contains information for webpack@4 and webpack@5? If yes I think we need note what previously this options had the `noEmitOnErrors` name

**Review by evilebottnawi — COMMENTED** [review]
```
 ✖ FAIL fragment-check dist/plugins/index.html --> /configuration/optimization/#optimizationnoemitonerrors
  | operator: fragment-check
  | expected: id="optimizationnoemitonerrors"
  |       at: dist/plugins/index.html:88:34 <a href="/configuration/optimization/#optimizationnoemitonerrors">...</a>
  
  ✖ FAIL fragment-check dist/plugins/printable/index.html --> /configuration/optimization/#optimizationnoemitonerrors
  | operator: fragment-check
  | expected: id="optimizationnoemitonerrors"
  |       at: dist/plugins/printable/index.html:3371:34 <a href="/configuration/optimization/#optimizationnoemitonerrors">...
```

**Comment by evilebottnawi** [comment]
Looks some pages still contain link on `optimizationnoemitonerrors`

**Comment by snitin315** [comment]
https://github.com/webpack/webpack.js.org/search?q=noEmitOnErrors&unscoped_q=noEmitOnErrors, it shows only the two places I have changed other than the code snippet.

**Review by evilebottnawi — COMMENTED** [review]
Need check page:
- content/plugins/index.html
- plugins/printable/index.html

Based on CI output

**Comment by chenxsan** [comment]
Search `optimizationnoemitonerrors` https://github.com/webpack/webpack.js.org/search?q=optimizationnoemitonerrors&unscoped_q=optimizationnoemitonerrors. This line https://github.com/webpack/webpack.js.org/blob/master/webpack.ssg.js#L79 contains `optimizationnoemitonerrors`, try to update it to see if it would fix CI.

File: src/content/configuration/optimization.md (hunk 3 above)
Comment (EugeneHlushko) [reply src/content/configuration/optimization.md:158]: Docs are v5, if something breaking is merged there is a v4 branch to do PRs against

**Comment by snitin315** [comment]
Its green now, Thanks !!

**Review by sokra — APPROVED** [review]

**Review by EugeneHlushko — APPROVED** [review]

**Comment by sokra** [comment]
Thanks

**Pull request merged**